be/src/olap/schema_change.cpp

```diff
@@ -1859,6 +1859,29 @@ OLAPStatus SchemaChangeHandler::_parse_request(TabletSharedPtr base_tablet,
     // 若Key列的引用序列出现乱序，则需要重排序
     int num_default_value = 0;
 
+    // A, B, C are keys(sort keys), D is value
+    // The following cases are not changing the order, no need to resort:
+    // (sort keys keep in same order)
+    //      old keys:    A   B   C   D
+    //      new keys:    A   X   B   C   D
+    //
+    //      old keys:    A   B   C   D
+    //      new keys:    X   A   B   C   D
+    //
+    //      old keys:    A   B   C   D
+    //      new keys:    A   B   C
+    //
+    //      old keys:    A   B   C   D
+    //      new keys:    A   B
```

Comment:
After this pull request.
old keys: A B C D
new keys: A B C
will be resorted.

Reply:
Sorry that this PR is not ready, I will consider more situation.

Reply:
This case also need be re-agg, so the sc_sorting must be marked as true.